### PR TITLE
loader/svg: Do recalc after gradient inheritance

### PR DESCRIPTION
**Pull request by JSUYA**
Inherit gradientUnits and recalculate each element
radial(cx, cy, r,...), linear(x1, y1, x2, y2).

related issue: https://github.com/thorvg/thorvg/issues/1255

### CODE REVIEW

**Comment by JSUYA** [comment]
The SVG_FILE_130758.svg problem in the regression test looks to be an inherit problem of the focal feature.
@mgrudzinska  Can you check this?

File: src/loaders/svg/tvgSvgLoader.cpp [hunk 1]
```diff
@@ -2669,11 +2669,13 @@ static void _inheritGradient(SvgLoaderData* loader, SvgStyleGradient* to, SvgSty
         if (to->transform) memcpy(to->transform, from->transform, sizeof(Matrix));
     }
 
-    if (to->type == SvgGradientType::Linear && from->type == SvgGradientType::Linear) {
```
Comment (hermet) [inline src/loaders/svg/tvgSvgLoader.cpp:2672]: This condition looks like somewhat unnecessary changes. It doesn't need to loop for().

Comment (JSUYA) [reply src/loaders/svg/tvgSvgLoader.cpp:2672]: from and to can be different gradients(linear, radial).
Regardless of what the from is, recalc must be performed according to changes in common properties.(unit, transform, spread)

Comment (hermet) [reply src/loaders/svg/tvgSvgLoader.cpp:2672]: @JSUYA please see 2675 line.

Comment (JSUYA) [reply src/loaders/svg/tvgSvgLoader.cpp:2672]: ok I removed 2675line if condition

File: src/loaders/svg/tvgSvgLoader.cpp [hunk 2]
```diff
@@ -2696,10 +2696,16 @@ static void _inheritGradient(SvgLoaderData* loader, SvgStyleGradient* to, SvgSty
             //GradUnits not set directly, coord set
             if (!gradUnitSet && coordSet) {
                 radialTags[i].tagRecalc(loader, to->radial, to->userSpace);
+                //If fx and fy are not set, set cx and cy.
+                if (!strcmp(radialTags[i].tag, "cx") && (to->flags & SvgGradientFlags::Cx) && !(to->flags & SvgGradientFlags::Fx)) to->radial->fx = to->radial->cx;
```
Comment (hermet) [inline src/loaders/svg/tvgSvgLoader.cpp:2700]: the center position is mandatory in radial gradient. In the exceptional case, it would overwrite with the default value anyway. Thus this condition looks unnecessary. 
``` !strcmp(radialTags[i].tag, "cx") && (to->flags & SvgGradientFlags::Cx) ```

**Review by hermet — COMMENTED** [review]
thanks. one more comment.

File: src/loaders/svg/tvgSvgLoader.cpp (hunk 2 above)
Comment (JSUYA) [reply src/loaders/svg/tvgSvgLoader.cpp]: um.. depending on the inheritance status, there may be a case where cx or cy is missing, but it will probably be overwritten with 0.
I removed `(to->flags & SvgGradientFlags::Cx)` condition.
`strcmp()` should be necessary. If not, it is overwritten at every tag, so recalculated fx or ,fy may be overwritten in `fr`.

**Review by hermet — APPROVED** [review]
